### PR TITLE
chore: rework trigger of Pull Build Image GitHub Action

### PR DESCRIPTION
**Pull request by the1bit**
**Description**

Changes proposed in this pull request:

- Modify the workflow file to exclude triggers for test folder updates, PR title changes, and .md file modifications

**Related issue(s)**

### CODE REVIEW

File: .github/workflows/pull-evaluation-tests.yaml [hunk 1]
```diff
@@ -10,14 +10,21 @@ run-name: "Evaluation tests"
...
     paths-ignore:
+      - ".github/**"
+      - ".reuse/**"
+      - "LICENSES/**"
+      - "config/**"
+      - "data/**"
       - "docs/**"
-      - "**.md"
+      - "scripts/**"
+      - "tests/**"
```
Comment (muralov) [inline .github/workflows/pull-evaluation-tests.yaml:24]: why are you excluding everything under the "tests" folder? The "blackbox" tests are under this folder. I think, if you exclude, the there will not be evaluation tests run if blackbox/evaluation tests are changed.

File: .github/workflows/pull-evaluation-tests.yaml [hunk 2]
```diff
@@ -10,14 +10,21 @@ run-name: "Evaluation tests"
 
 on:
   pull_request_target:
+    types: [labeled, opened, edited, synchronize, reopened, ready_for_review]
```
Comment (muralov) [inline .github/workflows/pull-evaluation-tests.yaml:13]: Can we delete `labeled` und `edited` event types? 
Labels: It seems the evaluation tests are triggered if we add/delete labels. 
Edit: triggers if description, title any other metadata is edited.

File: .github/workflows/pull-build-image.yaml [hunk 3]
```diff
@@ -3,6 +3,20 @@ name: Pull Build Image
 on:
   pull_request_target:
     types: [opened, edited, synchronize, reopened, ready_for_review]
```
Comment (muralov) [inline .github/workflows/pull-build-image.yaml:5]: Can we delete edited event type?
Edit: triggers if description, title any other metadata is edited.

File: .github/workflows/pull-evaluation-tests.yaml (hunk 1 above)
Comment (the1bit) [reply .github/workflows/pull-evaluation-tests.yaml:24]: This is the requirement in the the issue: https://github.com/orgs/kyma-project/projects/61/views/1?pane=issue&itemId=74792786&issue=kyma-project%7Ckyma-companion%7C136

"Modify the workflow file to exclude triggers for test folder updates, PR title changes, and .md file modifications"

File: .github/workflows/pull-evaluation-tests.yaml (hunk 2 above)
Comment (the1bit) [reply .github/workflows/pull-evaluation-tests.yaml:13]: labeled is required: https://github.com/kyma-project/kyma-companion/blob/6db22e4dfcf4562173ed08d505d08c962795b756/.github/workflows/pull-evaluation-tests.yaml#L53

**Review by muralov — APPROVED** [review]

**Pull request merged**